conn-sdk-cli/readmegen/testdata/test1want.md

```diff
@@ -0,0 +1,189 @@
+# Test
+
+Name: <!-- readmegen:name -->Test-Connector<!-- /readmegen:name -->
+Summary: <!-- readmegen:summary -->test summary<!-- /readmegen:summary -->
+Description: <!-- readmegen:description -->
+Test description
+should be able to handle new lines as well!
+<!-- /readmegen:description -->
+Version: <!-- readmegen:version -->v0.1.0<!-- /readmegen:version -->
+Author: <!-- readmegen:author -->test author<!-- /readmegen:author -->
+
+## Source Parameters
+
+<!-- readmegen:source.parameters.yaml -->
+```yaml
+version: 2.2
+pipelines:
+  - id: example
+    status: running
+    connectors:
+      - id: example
+        plugin: "test-connector"
+        settings:
+          # string param1 description
+          # Type: string
+          # Required
+          param1: "foo"
+          # int param2 description
+          # Type: int
+          # Optional
+          param2: "0"
+```
+<!-- /readmegen:source.parameters.yaml -->
+
+<!-- readmegen:source.parameters.table -->
+<table class="no-margin-table">
+  <tr>
+    <th>Name</th>
+    <th>Type</th>
+    <th>Required/Optional</th>
```

Comment:
Would it make sense to maybe just leave `Required` and the value would be `yes` or `blank`? What do you think? I personally think it's strange to have `Required / Optional` and only fill the value when it's required (and not optional). Having both would also make it harder to read I think.

Reply:
I wanted to have "required" and "optional" for two reasons: (1) when you're going through table and see "required" or "optional", it's pretty clear what it's about. If you see "yes" or "no", a bit less. (2) "required/optional" is a bit shorter then "required: yes" and "required: no".

Reply:
I agree with Raul, I feel like having both makes it harder for me to read at a glance. I'll make an even more controversial suggestion - how about we make use of some icons? Instead of this:
![Screenshot 2025-02-04 at 13 24 10](https://github.com/user-attachments/assets/fc83abd1-015d-4f3c-82e5-60b602c9f21f)

We'd have something like this:
![Screenshot 2025-02-04 at 13 25 42](https://github.com/user-attachments/assets/5fb6877b-be7e-4937-8607-e25302dfb64c)

To me the icons make required params stand out more 🤷

Reply:
I guess it's a matter of preference. I like reading a row as a sentence: param1 is a string, required, etc. But I can change this into a yes/no or an icon.